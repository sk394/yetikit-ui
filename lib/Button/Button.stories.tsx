import type { Meta, StoryObj } from '@storybook/react-vite';

import { Button } from './Button';

const meta = {
    component: Button,
    tags: ['autodocs'],
} satisfies Meta<typeof Button>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
    args: {
        children: 'Click Me',
    },
};

export const Emoji: Story = {
    args: {
        children: 'ðŸ˜¥',
    },
};
