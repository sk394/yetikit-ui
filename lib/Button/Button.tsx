import type { ReactNode } from 'react'

export interface ButtonProps {
    children: ReactNode
}

export const Button = ({ children }: ButtonProps): ReactNode => {
    return <button className="bg-red-100 text-lg uppercase">{children}</button>
}
