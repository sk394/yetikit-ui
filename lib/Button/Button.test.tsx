import { render } from '@testing-library/react'
import { Button } from './Button'

describe(`Component : ${Button.name}`, () => {
    it('should render correctly', () => {
        const { container } = render(<Button>Click me</Button>)
        expect(container).toMatchInlineSnapshot(`
      <div>
        <button
          class="bg-red-100 text-lg uppercase"
        >
          Click me
        </button>
      </div>
    `)
    })
})