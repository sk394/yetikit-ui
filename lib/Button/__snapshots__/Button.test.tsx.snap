// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component : Button > should render correctly 1`] = `
<div>
  <button
    class="bg-red-100 text-lg uppercase"
  >
    Click me
     Here we go again
  </button>
</div>
`;
